lib/utils/validate-password/index.ts

```diff
@@ -5,6 +5,6 @@ export const validatePassword = function (password: string, email: string) {
   }
   // minimum of 8 characters
   // allow symbols ~!@#$%^&*_-+=`|(){}[]:;"',.?/
-  const re = /^[a-zA-Z0-9~ !@#$%^&*_\-+=`|()[\]:;"',.?/]{8,}$/;
+  const re = /^[a-zA-Z0-9~ !@#$%^&*_\-+=`|(){}[\]:;"',.?/]{8,}$/;
```

Comment:
just for kicks and giggles, if our policy is "no newlines or tabs" then we can also encode that, which is more permissive for the full range of valid inputs

```ts
const re = /^[^\t\n\r]{8,}$/;
```

this can also simplify the validation if we also want to hard-code the rules in the clients (which I don't think we want to but…)

```ts
const digits = /\d/;
const symbols = /[~ !@$%^&*_\-+=`|(){}\[\]:;"',.?]/;

const isLong = re.test(password);
const hasDigits = digits.test(password);
const hasSymbols = symbols.test(password);

return isLong && hasDigits && hasSymbols
	? { isValid: true }
	: { isValid: false, needs: { digits: ! hasDigits, symbols: ! hasSymbols } }
```

anyhoo… :whistles: carry on 🥃

Reply:
https://github.com/Automattic/simplenote-electron/pull/2114